sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationPropertySource.java

```diff
@@ -87,9 +92,9 @@ public final class AppConfigurationPropertySource extends EnumerablePropertySour
 
     private final AppConfigurationProperties appConfigurationProperties;
 
-    private final HashMap<String, KeyVaultClient> keyVaultClients;
+    private final Map<String, KeyVaultClient> keyVaultClients;
```

Comment:
Seems like we will have multiple `AppConfigurationPropertySource` and each of them will create a bunch of KV clients?

Reply:
Can we delegate the KV to another component so we can delegate the KV functions to that component.

Reply:
I would rather not touch this at the moment, it was a change for this PR. It can be added to the list of 4.0 changes.